rustworkx-core/src/connectivity/find_cycle.rs

```diff
@@ -119,11 +126,49 @@ where
     cycle
 }
 
+fn find_node_in_arbitrary_cycle<G>(graph: G) -> Option<G::NodeId>
+where
+    G: GraphBase,
+    G: NodeCount,
+    G: EdgeCount,
+    for<'b> &'b G: GraphBase<NodeId = G::NodeId>
+        + IntoNodeIdentifiers
+        + IntoNeighborsDirected
+        + Visitable
+        + IntoEdgeReferences
+        + NodeIndexable,
+    G::NodeId: Eq + Hash,
+{
+    for scc in algo::tarjan_scc(&graph) {
```

Comment:
I remember we had panics or some issue with one of petgraph's strongly connected component functions along time ago. Was it kosaraju or tarjan?

Reply:
Switched to `kosaraju-scc` which is the one we use for the Python SCC function